parser-typechecker/src/Unison/Codebase/MainTerm.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Find a computation of type '{IO} () in the codebase.
+module Unison.Codebase.MainTerm where
+
+import Unison.Prelude
+
+import           Unison.Parser                  ( Ann )
+import qualified Unison.Parser                 as Parser
+import qualified Unison.Term                   as Term
+import           Unison.Var                     ( Var )
+import qualified Unison.DataDeclaration        as DD
+import qualified Unison.HashQualified          as HQ
+import qualified Unison.Referent               as Referent
+import qualified Unison.Names3                 as Names3
+import           Unison.Reference               ( Reference )
+import qualified Unison.Type                   as Type
+import           Unison.Type                    ( Type )
+import qualified Unison.Typechecker as Typechecker
+import           Unison.Runtime.IOSource        ( ioReference )
+
+data MainTerm v
+  = MtNotAFunctionName String
+  | MtNotFound String
+  | MtBadType String
+  | MtSuccess HQ.HashQualified (Term.AnnotatedTerm v Ann) (Type v Ann)
```

Comment:
Minor thing - could you remove the `Mt` prefix from constructor names. Convention we try to use is to import the module qualified so referencing the constructors is done via `MainTerm.NotAFunctionName`, etc. If we're using the names of a module a lot, we'll also sometimes give it shorter 1-2 letter qualified import, like just `MT`.